lib/bootloader_setup.pm

```diff
@@ -1040,37 +1040,90 @@ sub change_grub_config {
+    add_grub_cmdline_settings($add [, $update_grub , $search]);
 
 Add C<$add> into /etc/default/grub, using sed.
 C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+C<$search> if set, bypass default grub cmdline variable.
 =cut
 sub add_grub_cmdline_settings {
+    my $add  = shift;
+    my %args = compat_args(
+        {
+            add         => $add,
+            update_grub => 0,
+            search      => get_cmdline_var(),
+        }, ['add', 'update_grub', 'search'], @_);
+
+    change_grub_config('"$', " $add\"", $args{search}, "g", $args{update_grub});
+}
+
+=head2 add_grub_xen_cmdline_settings
+
+    add_grub_xen_cmdline_settings($add [, $update_grub ]);
+
+Add C<$add> into /etc/default/grub, using sed.
+C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+=cut
+sub add_grub_xen_cmdline_settings {
     my ($add, $update_grub) = @_;
-    change_grub_config('"$', " $add\"", get_cmdline_var(), "g", $update_grub);
+    add_grub_cmdline_settings($add, $update_grub, "GRUB_CMDLINE_XEN_DEFAULT");
 }
 
 =head2 replace_grub_cmdline_settings
 
-    replace_grub_cmdline_settings($old, $new [, $update_grub ]);
+    replace_grub_cmdline_settings($old, $new [, $update_grub , $search]);
 
 Replace C<$old> with C<$new> in /etc/default/grub, using sed.
 C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+C<$search> if set, bypass default grub cmdline variable.
 =cut
 sub replace_grub_cmdline_settings {
+    my $old  = shift;
+    my $new  = shift;
+    my %args = compat_args(
+        {
+            old         => $old,
+            new         => $new,
+            update_grub => 0,
+            search      => get_cmdline_var(),
+        }, ['old', 'new', 'update_grub', 'search'], @_);
+    change_grub_config($old, $new, $args{search}, "g", $args{update_grub});
+}
+
+=head2 replace_grub_xen_cmdline_settings
+
+    replace_grub_xen_cmdline_settings($old, $new [, $update_grub ]);
+
+Replace C<$old> with C<$new> in /etc/default/grub, using sed.
+C<$update_grub> if set, regenerate /boot/grub2/grub.cfg with grub2-mkconfig and upload configuration.
+=cut
+sub replace_grub_xen_cmdline_settings {
     my ($old, $new, $update_grub) = @_;
-    change_grub_config($old, $new, get_cmdline_var(), "g", $update_grub);
+    replace_grub_cmdline_settings($old, $new, $update_grub, "GRUB_CMDLINE_XEN_DEFAULT");
 }
 
 =head2 remove_grub_cmdline_settings
 
     remove_grub_cmdline_settings($remove);
 
 Remove C<$remove> from /etc/default/grub (using sed) and regenerate /boot/grub2/grub.cfg.
+Search line C<$search> from /etc/default/grub (use for sed).
 =cut
 sub remove_grub_cmdline_settings {
+    my ($remove, $search) = @_;
+    replace_grub_cmdline_settings('[[:blank:]]*' . $remove . '[[:blank:]]*', " ", "g", $search);
+}
+
+=head2 remove_grub_xen_cmdline_settings
+
+    remove_grub_xen_cmdline_settings($remove);
```

Comment:
`remove_grub_xen_cmdline_settings($remove, [ $search ]);`
\+ maybe note, that when `$search` is undef, it's used `replace_grub_cmdline_settings()` default (that's probably obvious).

Reply:
Let me fix them.

Reply:
looks like you haven't push the fix here

Reply:
@asmorodskyi I consider this PR ready, just nobody merged it (I don't have merge rights for openQA).